src/services/transcript-parser.js

```diff
@@ -18,114 +18,132 @@ const TRANSCRIPT_MIME_TYPES = [
-              let supplementingItems = annotation.getBody();
-              supplementingItems.forEach((si, subIndex) => {
-                let label = si.getLabel()[0] ? si.getLabel()[0].value : `${i}`;
-                let id = si.id;
+      if (canvases?.length > 0) {
+        canvases.map((canvas, index) => {
+          let annotations = parseAnnotations(canvas.__jsonld['annotations'], 'supplementing');
+          let canvasTranscripts = [];
+          if (annotations.length > 0) {
+            let annotBody = annotations[0].getBody()[0];
+            if (annotBody.getProperty('type') === 'TextualBody') {
+              let label = title.length > 0
+                ? title
+                : (annotBody.getLabel()[0]
+                  ? annotBody.getLabel()[0].value
+                  : `Canvas-${index}`
+                );
+              let { isMachineGen, labelText } = identifyMachineGen(label);
+              canvasTranscripts.push({
+                url: annotBody.id === undefined ? manifestURL : annotBody.id,
+                title: labelText,
+                isMachineGen: isMachineGen,
+                id: `${labelText}-${index}`,
+              });
+            } else {
+              annotations.forEach((annotation, i) => {
+                let annotBody = annotation.getBody()[0];
+                let label = annotBody.getLabel() != undefined ? annotBody.getLabel()[0].value : `${i}`;
+                let id = annotBody.id;
                 let sType = identifySupplementingAnnotation(id);
                 let { isMachineGen, labelText } = identifyMachineGen(label);
                 if (sType === 1 || sType === 3) {
-                  newTranscriptsList.push({
-                    title: title.length > 0 ? `${title} - ${labelText}` : labelText,
+                  canvasTranscripts.push({
+                    title: labelText,
                     url: id,
-                    validity: TRANSCRIPT_VALIDITY.transcript,
-                    isMachineGen: item.isMachineGen || isMachineGen,
-                    id: `${title}-${i}-${index}-${subIndex}`
+                    isMachineGen: isMachineGen,
+                    id: `${labelText}-${index}-${i}`
                   });
                 }
               });
-            });
+            }
           }
-        } else {
-          newTranscriptsList.push({ ...item, validity: TRANSCRIPT_VALIDITY.noTranscript });
-        }
-      } else {
-        newTranscriptsList.push({ ...item, validity: TRANSCRIPT_VALIDITY.transcript });
+          newTranscriptsList.push({ canvasId: index, items: canvasTranscripts });
+        });
       }
       return newTranscriptsList;
     })
-    .catch(function () {
-      return [{ ...item, validity: TRANSCRIPT_VALIDITY.invalidURL }];
+    .catch(function (err) {
+      console.error('Error fetching manifest, ', manifestURL);
+      return [];
     });
   return data;
 }
 
+export async function sanitizeTranscripts(transcripts) {
```

Comment:
I'm a bit confused as to what all this method is doing.  Could you please add a JSDoc method description comment and/or break it down into smaller methods?

Reply:
I was not able to break it down to smaller methods without breaking the code :disappointed:. 

So I added comments to it as much as I could to explain what each code block does.